### ISSUE TITLE
feat(py): 組み込みPython面のベクトルAPIパリティ（get_vectors の未公開、バッチupsert、重複ID原子拒否）

### ISSUE THREAD

**Issue by asopitech** [alopex-db/alopex#450]
## 概要

組み込みPythonバインディングのKV系ベクトルAPIに、Rust側やサーバ側にはあるものが
いくつか出ていません。性能改善の要望ではなく（それは #447 / #448）、
API表面のパリティの話です。

## 1. `get_vectors`（バッチ読み出し）がPythonに出ていない

Rust側の組み込みAPIには存在します:

```rust
// crates/alopex-tools/src/bin/demo_v08_embedded.rs:1026
let batch = txn.get_vectors(&[b"a".to_vec(), b"missing".to_vec()], metric)?;
```

一方 Python 側は単体の `get_vector` のみです:

```python
>>> [m for m in dir(alopex.Transaction) if "vector" in m or "hnsw" in m]
['delete_from_hnsw', 'get_vector', 'search_similar', 'upsert_to_hnsw', 'upsert_vector']
```

`_alopex.pyi` にも `get_vectors` はありません。

## 2. `upsert_to_hnsw` のバッチ版が無い

```python
# 現在の唯一の形
with db.begin(TxnMode.READ_WRITE) as txn:
    for i in range(n):
        txn.upsert_to_hnsw("idx", str(i).encode(), emb[i], None)
    txn.commit()
```

呼び出し側はすでに `(n, dim)` の NumPy 配列を持っているので、そのまま渡せる形が自然です。

```python
def upsert_to_hnsw_batch(
    self,
    name: str,
    keys: Sequence[bytes],
    vectors: Any,                      # numpy.ndarray[float32], shape (n, dim), C-contiguous
    metadata: Optional[Sequence[Optional[bytes]]] = None,
) -> int: ...
```

既存の `vector::with_ndarray_f32_gil_safe` で2次元float32配列をゼロコピー受け取りし、
1件ごとではなくバッチ全体で1回だけ `py.detach()` する形を想定しています。

**速度が目的ではないことを明記しておきます。** 実測では
`Transaction.put()` が1回あたり約1.1 µs で、`upsert_to_hnsw` 1回のコストの約0.01%でした。
またSQLの複数行INSERT（＝すでに存在するバッチ経路）でも構築時間はほぼ変わりません（#447）。
したがって本項の根拠は下の3点です。

## 3. #411 の原子的セマンティクスが組み込み面に無い

#411 で追加されたサーバ側 `upsert_batch_impl` は
「空バッチの拒否」「バッチ内ID重複の原子的な拒否（一件も適用しない）」を保証しますが、
組み込み面には対応物がなく、サーバ利用者と組み込み利用者に同じ契約を提示できません。

## 根拠

1. **原子性のパリティ** — 上記3。
2. **Rust ⇄ Python のパリティ** — `get_vectors` はRustにあってPythonに無い。
3. **エコシステムの標準形** — 主要なベクトルDBのPythonクライアントは、いずれも
   バッチが第一級で、1件ずつ専用のAPIを持ちません（各パッケージをインストールして
   `inspect.signature` で実シグネチャを確認）。

   | 製品 | バッチ投入API |
   |---|---|
   | qdrant-client 1.19.1 | `upsert(collection_name, points: Batch \| Sequence[PointStruct])`、`upload_collection(vectors=...)`、`upload_points(..., batch_size=64, parallel=1)` |
   | pymilvus 3.0.2 | `insert(collection_name, data: Dict \| List[Dict])` / `upsert(...)` 同形 |
   | chromadb 1.5.9 | `Collection.add(ids, embeddings: ndarray \| List[ndarray], ...)` / `upsert(...)` 同形 |
   | lancedb 0.39.0 | `Table.add(data)` / `merge_insert(on)` ビルダ |

   Chroma と LanceDB は alopex と同じく組み込み型で、それでもバッチが基本形です。

## 補足: バッチ経路自体はSQL側に既に存在します

組み込みPythonからでも、SQL経由なら複数行まとめて投入できます。

```python
db.execute_sql("CREATE TABLE items (id INTEGER PRIMARY KEY, embedding VECTOR(128, COSINE))")
db.execute_sql("CREATE INDEX idx_emb ON items (embedding) USING HNSW")
db.execute_sql("INSERT INTO items (id, embedding) VALUES (0,[...]),(1,[...]), ...")
```

制約として、MessagePack AST の上限（1文あたり65,536値 / 1 MiB）があるため、
128次元なら1文あたり概ね200〜500行が上限です。上限を超えると次のエラーになります。

```
error[ALOPEX-P001]: unexpected token: expected bounded MessagePack AST matching
docs/ffi-ast-contract.md, found MessagePack collection limit of 65536 values exceeded
```

本issueで求めているのは、このSQL経路とは別の、KV/HNSW系APIの表面の整合です。

環境: `alopex==0.8.13`（PyPI wheel）、Python 3.11、Linux x86_64、numpy 2.x

**Comment by asopitech** [comment]
Implemented in e17d3f4c: embedded Python now exposes get_vectors() and upsert_to_hnsw_batch(); batches reject empty/mismatched inputs and duplicate IDs before graph mutation. Added regression coverage and updated stubs. Full workspace clippy passes.

**Comment by asopitech** [comment]
Codexはコミット 2f086b59 で、C-contiguous float32 の2次元ndarrayをコピーせずにバッチupsertへ渡し、GIL解放は1回に限定しました。非連続・型変換が必要な入力は所有コピーへフォールバックします。CodexはPython実行テストで、キー/ベクトル数不一致が例外となりHNSW状態を変更しないことを確認しました（2 passed）。

**Comment by asopitech** [comment]
Codexは、get_vectors、バッチHNSW upsert、重複IDの原子拒否を実装したコミット2f086b59がv0.8.14／v0.8.15に含まれることを確認しました。公開wheelシナリオも成功しているため、完了とします。
